go/pumiceDB/client/PumiceDBClient.go

```diff
@@ -281,38 +271,46 @@ func (pmdb_client *PmdbClientObj) PmdbGetLeader() (uuid.UUID, error) {
 // If application expects response on write operation,
 // get_response should be 1
 func (obj *PmdbClientObj) writeKV(rncui string, key *C.char,
-	keyLen int64, get_response C.int,
-	replySize *int64) (unsafe.Pointer, error) {
+    keyLen int64, get_response C.int,
+    replySize *int64, reqArgs *PmdbReqArgs) error {
 
-	var obj_stat C.pmdb_obj_stat_t
+    var obj_stat C.pmdb_obj_stat_t
 
-	rncuiStrC := GoToCString(rncui)
-	defer FreeCMem(rncuiStrC)
+    rncuiStrC := GoToCString(rncui)
+    defer FreeCMem(rncuiStrC)
 
-	keyLenC := GoToCSize_t(keyLen)
+    keyLenC := GoToCSize_t(keyLen)
 
-	var rncui_id C.struct_raft_net_client_user_id
+    var rncui_id C.struct_raft_net_client_user_id
 
-	C.raft_net_client_user_id_parse(rncuiStrC, &rncui_id, 0)
-	var obj_id *C.pmdb_obj_id_t
+    C.raft_net_client_user_id_parse(rncuiStrC, &rncui_id, 0)
+    var obj_id *C.pmdb_obj_id_t
 
-	obj_id = (*C.pmdb_obj_id_t)(&rncui_id.rncui_key)
+    obj_id = (*C.pmdb_obj_id_t)(&rncui_id.rncui_key)
 
-	rc := C.PmdbObjPut(obj.pmdb, obj_id, key, keyLenC, get_response,
-		&obj_stat)
+    rc := C.PmdbObjPut(obj.pmdb, obj_id, key, keyLenC, get_response,
+        &obj_stat)
 
-	if rc != 0 {
-		return nil, fmt.Errorf("PmdbObjPut(): %d", rc)
-	}
+    if rc != 0 {
+        return fmt.Errorf("PmdbObjPut(): %d", rc)
+    }
 
-	get_response_go := int(get_response)
-	if get_response_go == 1 {
-		reply_buf := obj_stat.reply_buffer
-		*replySize = int64(obj_stat.reply_size)
-		return reply_buf, nil
-	}
+    get_response_go := int(get_response)
+    if get_response_go == 1 {
+        replySizeVal := int64(obj_stat.reply_size)
+
+        // Convert the unsafe.Pointer to a []byte
+        replyBytes := C.GoBytes(obj_stat.reply_buffer, C.int(replySizeVal))
+
+	buffer := bytes.NewBuffer(replyBytes)
+	*reqArgs.Response = buffer.Bytes()
+	
+	// Free the buffer allocated by the C library
+	C.free(obj_stat.reply_buffer)
+
```

Comment:
Do we need to copy the buffer here?  Prior to this patch the pointer was being returned?  Can we not store the reply pointer in `PmdbReqArgs` and release it when `PmdbReqArgs` is freed?

Reply:
Done